lib/GHCup/Utils/Pager.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE RankNTypes  #-}
+
+module GHCup.Utils.Pager where
+
+import GHCup.Prelude
+import System.Environment
+import GHCup.Utils.Dirs (findExecutable)
+import Data.Foldable (asum)
+import System.Process
+import System.Exit
+import System.IO
+import Data.Text (Text)
+import qualified Data.Text.IO as T
+import Control.Monad (forM_)
+import Control.Exception (IOException, try)
+import GHCup.Utils.Output
+import qualified Data.Text as T
+
+
+getPager :: IO (Maybe FilePath)
+getPager = do
+  lookupEnv "GHCUP_PAGER" >>= \case
+    Just r  -> pure $ Just r
+    Nothing -> lookupEnv "PAGER" >>= \case
+      Just r' -> pure $ Just r'
+      Nothing -> do
+        let pagers
+              | isWindows = ["most.exe", "more.exe", "less.exe"]
+              | otherwise = ["most", "more", "less"]
+        asum $ fmap findExecutable pagers
+
+sendToPager :: FilePath -> [Text] -> IO (Either IOException ())
+sendToPager pager text = try @IOException
+    $ withCreateProcess (shell pager) { std_in = CreatePipe
+                                      , std_err = CreatePipe
+                                      , std_out = Inherit
+                                      , delegate_ctlc = True
+                                      }
+    $ \(Just stdinH) _ (Just stderrH) ph -> do
+        forM_ text $ T.hPutStrLn stdinH
+        hClose stdinH
+        exitCode <- waitForProcess ph
+        case exitCode of
+          ExitFailure i ->
+            do errContents <- hGetContents stderrH
+               fail (unlines [mappend "Pager exited with exit code " (show i)
+                             ,errContents])
+          _ -> pure ()
```

Comment:
I looked into the `more` support, by looking at its source code. It reads from the STDERR and thus gets stuck if we create a new pipe for std_err. I confirmed that this change fixes `more` on both linux and windows
 https://github.com/util-linux/util-linux/blob/81ea117e0396274808a77d40bf11af44ea047285/text-utils/more.c#L804C11-L804C24

```suggestion
-- 'more' reads from STDERR, and requires std_err to be 'Inherit'
sendToPager :: FilePath -> [Text] -> IO (Either IOException ())
sendToPager pager text = try @IOException
    $ withCreateProcess (shell pager) { std_in = CreatePipe
                                      , std_err = Inherit
                                      , std_out = Inherit
                                      , delegate_ctlc = True
                                      }
    $ \(Just stdinH) _ _ ph -> do
        forM_ text $ T.hPutStrLn stdinH
        hClose stdinH
        exitCode <- waitForProcess ph
        case exitCode of
          ExitFailure i -> fail ("Pager exited with exit code " <> show i)
          _ -> pure ()
```

Reply:
Great find